core/src/main/java/com/linecorp/armeria/unsafe/ByteBufHttpData.java

```diff
@@ -62,18 +62,13 @@ public boolean isEndOfStream() {
 
     @Override
     public byte[] array() {
-        if (buf.hasArray()) {
+        if (buf.hasArray() && buf.arrayOffset() == 0) {
             return buf.array();
         } else {
             return ByteBufUtil.getBytes(buf);
```

Comment:
I'm curious if we have to cache the array. What do you think?

Reply:
I think it's uncommon enough to call it twice that it's ok without. If I'm not mistaken adding a fourth field to this class will tip it into being a larger object size so memory usage of all apps, even if they don't use this method, will go up.